modules/concurrency/src/spinner.cpp

```diff
@@ -79,12 +89,25 @@ auto Spinner::stop() -> std::future<void> {
   stop_requested_.store(true);
   condition_.notify_all();
 
-  return std::async(std::launch::async, [this]() { stopImpl(); });
+  auto stop_future = std::async(std::launch::async, [this]() { stopImpl(); });
+
+  if (async_spinner_handle_.valid()) {
+    try {
+      exception_thrown_in_spin_promise_.get_future().get();  // Check for any exception thrown during spin().
+    } catch (const std::exception& e) {
+      throw e;  // Re-throw the exception to be handled by the caller.
+    }
+  }
+
+  return stop_future;
```

Comment:
I think we can replace all of this with just 
```suggestion
  return std::move(async_spinner_handle_);
```
and remove `stopImpl`